synthesizer/process/src/stack/finalize_types/initialize.rs

```diff
@@ -483,6 +484,35 @@ impl<N: Network> FinalizeTypes<N> {
         Ok(())
     }
 
+    /// Ensures the given `metadata.get` command is well-formed.
+    #[inline]
+    fn check_metadata_get(
+        &mut self,
+        stack: &(impl StackMatches<N> + StackProgram<N>),
+        metadata_get: &MetadataGet<N>,
+    ) -> Result<()> {
+        // Ensure that the program is defined.
+        // Note that we do not check that the metadata name is valid, as that can change by an update to the program.
+        if let Some(program_id) = metadata_get.program() {
+            // Ensure the program ID is imported by the current program.
+            if !stack.program().imports().keys().contains(program_id) {
+                bail!("External program '{program_id}' is not imported by '{}'.", stack.program_id());
+            }
+            // Check that the process contains the external program.
+            stack.get_external_stack(program_id)?;
+        }
+        // Get the destination register.
+        let destination = metadata_get.destination().clone();
+        // Ensure the destination register is a locator (and does not reference an access).
+        ensure!(matches!(destination, Register::Locator(..)), "Destination '{destination}' must be a locator.");
+        // Insert the destination register.
+        self.add_destination(
+            destination,
+            FinalizeType::Plaintext(PlaintextType::Literal(*metadata_get.destination_type())),
```

Comment:
Are we or should we be constraining the destination type anywhere?

Reply:
`destination_type()` returns a `LiteralType`
Is there a particular constraint you had in mind?